src/Microsoft.TemplateEngine.Utils/ParserExtensions.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.TemplateEngine.Utils
+{
+    public static class ParserExtensions
+    {
+        /// <summary>
+        /// <see cref="double.TryParse(string, out double)"/> extension that try to parse <paramref name="stringValue"/> first in current culture, then in invariant culture.
+        /// </summary>
+        /// <param name="stringValue">value to parse</param>
+        /// <param name="doubleValue">parsed double value if <paramref name="stringValue"/> can be parsed</param>
+        /// <returns>
+        /// true when <paramref name="stringValue"/> can be parsed in current or invariant culture.
+        /// false when <paramref name="stringValue"/> cannot be parsed in current or invariant culture.
+        /// </returns>
+        public static bool DoubleTryParseСurrentOrInvariant(string stringValue, out double doubleValue)
+        {
+            if (double.TryParse(stringValue, out doubleValue))
+            {
+                return true;
+            }
+            else if (double.TryParse(stringValue, NumberStyles.Any, CultureInfo.InvariantCulture, out doubleValue))
+            {
+                return true;
+            }
```

Comment:
Why do we try two different locales instead of just using `InvariantCulture`?

Reply:
We discussed to keep current culture for compatibility. Current logic works correctly only if double is specified in current culture. It might be that someone is using it as the workaround to get template working.

Reply:
This wouldn't work consistently across different cultures though, right?

For instance, lets say a template has the following comparison ` < 3.1`.
In Turkish locale (and likely in some others) this will be evaluated to ` < 31` while in almost all other locales, it will be ` < 3.1`. This won't manifest itself as an error and will silently evaluate to a wrong value. 
I think this makes it more dangerous than before as it hides the problem.

I understand that it won't be backwards compatible, but always using invariant culture may be the right way to approach this, in my opinion. In C#, for instance, people don't code in their own locale: it is `double d = 3.4;` and never `double d = 3,4;`. We could follow a similar approach.

Reply:
Yes, you are correct for countries which uses '.' as decimal separator (as Turkey).  In this case I can only suggest to limit parsing to float style numbers only to solve it.  This will cover the cases where the users could use "," as decimal separator, however will avoid parsing doubles for "," for cultures where "." is a decimal separator, and "," is a group separator.